# origin: Aharon Robbins debian bts #147469
# cset: 416525d9BLl5Q1YCwhVzXQxKda4wYw
# inclusion: backport 2.4.28-pre4
# revision date: 2004-11-18
# description:  menuconfig fix crash due to infinite recursion
#              
--- a/scripts/Menuconfig	2004-11-17 18:43:29 -08:00
+++ b/scripts/Menuconfig	2004-11-17 18:43:29 -08:00
@@ -711,7 +711,7 @@
 
 function parser(ifile,menu) {
 
-	while (getline <ifile) {
+	while ((getline <ifile) > 0) {
 		if ($1 == "mainmenu_option") {
 			comment_is_option = "1"
 		}
@@ -758,7 +758,7 @@
 
 function parser(ifile,menu) {
 
-	while (getline <ifile) {
+	while ((getline <ifile) > 0) {
 		if ($0 ~ /^#|$MAKE|mainmenu_name/) {
 			printf("") >>menu
 		}
